backend/pkg/schema/service.go

```diff
@@ -224,3 +229,52 @@ func (s *Service) GetConfig() (*ConfigResponse, error) {
 func (s *Service) GetSubjectConfig(subject string) (*ConfigResponse, error) {
 	return s.registryClient.GetSubjectConfig(subject)
 }
+
+// ParseAvroSchemaWithReferences parses an avro schema that potentially has references
+// to other schemas. References will be resolved by requesting and parsing them
+// recursively. If any of the referenced schemas can't be fetched or parsed an
+// error will be returned.
+func (s *Service) ParseAvroSchemaWithReferences(schema *SchemaResponse) (avro.Schema, error) {
+	if len(schema.References) == 0 {
+		return avro.Parse(schema.Schema)
+	}
+
+	// Fetch and parse all schema references recursively. All schemas that have
+	// been parsed successfully will be cached by the avro library.
+	for _, reference := range schema.References {
+		schemaRef, err := s.GetSchemaBySubjectAndVersion(reference.Subject, strconv.Itoa(reference.Version))
+		if err != nil {
+			return nil, err
+		}
+
+		if _, err := s.ParseAvroSchemaWithReferences(&SchemaResponse{
+			Schema:     schemaRef.Schema,
+			SchemaType: schemaRef.Type,
+			References: schemaRef.References,
+		}); err != nil {
+			return nil, fmt.Errorf(
+				"failed to parse schema reference (subject: %q, version %q): %w",
+				reference.Subject, reference.Version, err,
+			)
+		}
+	}
+
+	// Parse the main schema in the end after solving all references
+	return avro.Parse(schema.Schema)
```

Comment:
@igormq Can you review this and possibly test this against a topic that uses avro references?

Reply:
Doing a test now, once sec

Reply:
<img width="587" alt="image" src="https://user-images.githubusercontent.com/623536/227925876-178e6ba8-5303-468d-858e-e119056768a3.png">
working :)

Reply:
Also, if it is not to ask much, once this pr is merged. can we have a new release (and docker image) available in your side?

Reply:
Thanks for testing. A new release has to wait until the new feature release (~1-2 more months), but docker images are built for every push. So once merged it will take ~10minutes until a new Docker image is built and pushed to this repo: https://hub.docker.com/r/redpandadata/console-unstable/tags